wire-ios-sync-engine/Source/Repositories/UserModelMappings.swift

```diff
@@ -0,0 +1,62 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+import WireAPI
+
+extension Collection where Element == WireDataModel.QualifiedID {
```

Comment:
As long as these mapping helpers stay `internal`, I'm ok with it.
For public scope I would rather repeat the mapping in place to make the legacy issues visible.

Reply:
The mapping should stay internal.

For base structs like `QualifiedID` I don't think we need separate type in each layer we could share them. The same way we don't have separate types for String and Int in each layer.

Reply:
I agree mostly, just keep in mind that this create a dependency to a module where `QualifiedID` is defined.
One could accept it for few types and live with it or think about a refactoring, e.g. introducing a light-weight module `WireDataTypes` or `WireDomainTypes` etc...

Reply:
One other thing could be to split feature wise, e.g. `WireSearch` that imports `WireSearchTypes`. Just my thoughts, nothing to change in this PR 👍

Reply:
Perhaps it could be `WireFoundation` and remain lightweight?